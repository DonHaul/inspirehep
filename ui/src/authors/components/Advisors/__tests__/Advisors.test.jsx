import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';
import { MemoryRouter } from 'react-router-dom';
import Advisors from '../Advisors';

describe('Advisors', () => {
  it('renders advisors', () => {
    const advisors = fromJS([
      {
        name: 'Degreeless Advisor',
      },
      {
        name: 'Other Advisor',
        degree_type: 'other',
      },
      {
        name: 'Master Advisor',
        degree_type: 'master',
      },
      {
        name: 'PhD Advisor',
        degree_type: 'phd',
      },
      {
        name: 'Another PhD Advisor',
        degree_type: 'phd',
      },
    ]);
    const { asFragment } = render(
      <MemoryRouter>
        <Advisors advisors={advisors} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
