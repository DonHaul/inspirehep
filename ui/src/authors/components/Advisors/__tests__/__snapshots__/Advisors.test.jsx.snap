// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advisors renders advisors 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
  >
    <span
      class=""
    >
      Advisors: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=Degreeless%20Advisor"
        >
          Degreeless Advisor
        </a>
        , 
      </li>
      <li>
        <a
          href="/authors?q=Other%20Advisor"
        >
          Other Advisor
        </a>
      </li>
    </ul>
  </div>
  <div
    class="__InlineList__"
  >
    <span
      class=""
    >
      Master Advisor: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=Master%20Advisor"
        >
          Master Advisor
        </a>
      </li>
    </ul>
  </div>
  <div
    class="__InlineList__"
  >
    <span
      class=""
    >
      PhD Advisors: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=PhD%20Advisor"
        >
          PhD Advisor
        </a>
        , 
      </li>
      <li>
        <a
          href="/authors?q=Another%20PhD%20Advisor"
        >
          Another PhD Advisor
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
